Mathlib/Algebra/Order/Quantale.lean

```diff
@@ -71,44 +71,45 @@ overhead if a user does not need them.
 
 open Function
 
+namespace Quantale
+
 /-- An additive quantale is an additive semigroup distributing over a complete lattice. -/
-class AddQuantale (α : Type*) [AddSemigroup α] extends CompleteLattice α where
+class IsAddQuantale (α : Type*) [AddSemigroup α] [CompleteLattice α] where
   /-- Addition is distributive over join in a quantale -/
   protected add_sSup_eq_iSup_add (x : α) (s : Set α) : x + sSup s = ⨆ y ∈ s, x + y
   /-- Addition is distributive over join in a quantale -/
   protected sSup_add_eq_iSup_add (s : Set α) (y : α) : sSup s + y = ⨆ x ∈ s, x + y
 
 /-- A quantale is a semigroup distributing over a complete lattice. -/
 @[to_additive]
-class Quantale (α : Type*) [Semigroup α] extends CompleteLattice α where
+class IsQuantale (α : Type*) [Semigroup α] [CompleteLattice α] where
   /-- Multiplication is distributive over join in a quantale -/
   protected mul_sSup_eq_iSup_mul (x : α) (s : Set α) : x * sSup s = ⨆ y ∈ s, x * y
   /-- Multiplication is distributive over join in a quantale -/
   protected sSup_mul_eq_iSup_mul (s : Set α) (y : α) : sSup s * y = ⨆ x ∈ s, x * y
 
 /-- An integral (or strictly two-sided) additive quantale is a quantale over an additive monoid
 `⊤` and `0` coincide. -/
-class IsIntegralAddQuantale (α : Type*) [AddMonoid α] [AddQuantale α] : Prop where
+class IsIntegralAddQuantale (α : Type*) [AddMonoid α] [CompleteLattice α] [IsAddQuantale α] : Prop
+  where
```

Comment:
```suggestion
class IsIntegralAddQuantale (α : Type*) [AddMonoid α] [CompleteLattice α] extends IsAddQuantale α where
```
since `IsIntegralAddQuantale` belonds to the same hierarchy as `IsAddQuantale`